### ISSUE TITLE
I'm glad to see this reworking.

### ISSUE THREAD

**Issue by TQRTQ** [openclimatefix/skillful_nowcasting#9]
I am a graduate student majoring in deep learning. I am also reproducing this Nature paper and have finished the code work. I want to talk to you about the code. This is my code address：https://github.com/TQRTQ/DGMR

**Comment by jacobbieker** [comment]
Nice! Your implementation looks quite good, does it train correctly? This repo's implementation still gives NaN values in the Sampler.

**Comment by TQRTQ** [comment]
I have not made use of the data provided in the paper to conduct experiments, and I have encountered some difficulties in processing the data in the paper. So I randomly initialized some data, and the code worked fine.
But tention is my spatial intention, you is self-attention. I'll read your code carefully and learn about it. In addition, I would like to ask you about data preprocessing

**Comment by jacobbieker** [comment]
Yeah, they released the pseudocode, which is now also in this repo. So my attention is copied from theirs. Currently, I haven't worked with their data yet, so its just been random data in the unit tests. I'll probably start with their data soonish though to make sure it works, and with porting over their weights.
